opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -935,7 +936,15 @@ computeNetworkPressures(const Network::ExtNetwork& network,
 #endif
                 } else {
                     // Table number specified as 9999 in the deck, no pressure loss.
-                    node_pressures[node] = up_press;
+                    if (network.node(node).as_choke()){
+                        // Node pressure is set to the common group THP of the wells.
+                        // The choke pressure must be non-negative therefore the node pressure of 
+                        // the auto-choke node must be larger or equal to the pressure of the uptree node of its branch 
+                        const auto group_thp = group_state.well_group_thp(node);
+                        node_pressures[node] = group_thp >= up_press ? group_thp : up_press;
```

Comment:
We had some similar logic in the following code, 
```c++
                 if (autochoke_thp.has_value()) {
                    well_group_thp_calc_[nodeName] = autochoke_thp.value();
                    well_group_thp = std::max(autochoke_thp.value(), nodal_pressure);
                }
```
Do we need to enforce this logic again?

Reply:
It is indeed redundant. I removed it.